packages/perf/test/index.spec.ts

```diff
@@ -0,0 +1,108 @@
+  maxInboundStreams: 1 << 10,
+  maxOutboundStreams: 1 << 10,
+  timeout: 10000,
+  writeBlockSize: BigInt(64 << 10)
+}
+
+async function createComponents (): Promise<Components> {
+  const peerId = await createEd25519PeerId()
+
+  const events = new EventEmitter()
+
+  const components = defaultComponents({
+    peerId,
+    registrar: mockRegistrar(),
+    upgrader: mockUpgrader(),
+    datastore: new MemoryDatastore(),
+    transportManager: stubInterface<TransportManager>(),
+    connectionGater: mockConnectionGater(),
+    events
+  })
+
+  components.peerStore = new PersistentPeerStore(components)
+  components.connectionManager = new DefaultConnectionManager(components, {
+    minConnections: 50,
+    maxConnections: 1000,
+    autoDialInterval: 1000,
+    inboundUpgradeTimeout: 1000
+  })
+
+  return components
+}
+
+describe('perf', () => {
+  let localComponents: Components
+  let remoteComponents: Components
+
+  beforeEach(async () => {
+    localComponents = await createComponents()
+    remoteComponents = await createComponents()
+
+    await Promise.all([
+      start(localComponents),
+      start(remoteComponents)
+    ])
+  })
+
+  afterEach(async () => {
+    await Promise.all([
+      stop(localComponents),
+      stop(remoteComponents)
+    ])
+  })
+
+  it('should run perf', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    // Run Perf
+    await expect(client.perf(remoteComponents.peerId, 1n << 10n, 1n << 10n)).to.eventually.be.fulfilled()
+  })
+
+  it('should output benchmark', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    // Run Perf
+    const downloadBandwidth = await client.measureDownloadBandwidth(remoteComponents.peerId, 10n << 20n) >> 10
+    // eslint-disable-next-line no-console
+    console.log('Download bandwidth:', downloadBandwidth, 'kiB/s')
+
+    const uploadBandwidth = await client.measureDownloadBandwidth(remoteComponents.peerId, 10n << 20n) >> 10
+    // eslint-disable-next-line no-console
+    console.log('Upload bandwidth:', uploadBandwidth, 'kiB/s')
```

Comment:
In `measureDownloadBandwidth` you use power of 10, e.g. `8000`, whereas here you use power of 2, e.g. `10n << 20n`. I suggest being consistent, that is use either but not both.